src/Raven.Studio/typescript/common/queryCompleter.ts

```diff
@@ -236,23 +239,34 @@ class queryCompleter {
                     result.parentheses--;
                     break;
                 case "space":
-                    if (!result.keyword) {
+                    if (!result.keyword && !isBeforeComma) {
                         if (!lastToken || lastToken.type !== "space") {
                             result.dividersCount++;
                         }
+
+                        if (isFieldPrefixMode === 1) {
+                            isFieldPrefixMode = 2;
```

Comment:
It is magic constant ;-) I have no idea what isFieldPrefixMode do. Can we use string enums instead?

Reply:
This part would be re-worked as part of the following task:
http://issues.hibernatingrhinos.com/issue/RavenDB-8184